tools/modules/AksEdgeDeploy/AksEdgeDeploy.psm1

```diff
@@ -280,7 +280,10 @@ function UpgradeJsonFormat {
     #upgrade from public preview format to GA format
     $edgeCfg = $jsonObj.AksEdgeConfig
 
-    if ($edgeCfg.SchemaVersion -gt "1.4") {
+    $version = [int[]]$edgeCfg.SchemaVersion.Split(".")
+    $majorVersion = [int]$version[0]
+    $minorVersion = [int]$version[1]
+    if (($majorVersion -gt 1) -or ($majorVersion -eq 1 -and $minorVersion -gt 4)) {

```

Comment:
nit: you may want to use System.Version for generic version comparison logic: https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-version

Reply:
used System.Version